control-plane/connect-inject/peering_dialer_controller_test.go

```diff
@@ -502,10 +504,24 @@ func TestReconcileDeletePeeringDialer(t *testing.T) {
 	for _, tt := range cases {
 		t.Run(tt.name, func(t *testing.T) {
 			// Add the default namespace.
-			ns := corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "default"}}
+			ns := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "default"}}
+
+			dialer := &v1alpha1.PeeringDialer{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:              "dialer-deleted",
+					Namespace:         "default",
+					DeletionTimestamp: &metav1.Time{Time: time.Now()},
+					Finalizers:        []string{FinalizerName},
+				},
+				Spec: v1alpha1.PeeringDialerSpec{
+					Peer: &v1alpha1.Peer{
+						Secret: nil,
+					},
+				},
+			}
 
```

Comment:
Similar to my comment above, could we make an assertion on the finalizer key being removed?

Reply:
So the object gets deleted from the cluster but I wrote a test to validate that.

Reply:
Perfect!